### DISCUSSION TITLE
toggle element option in documents

### DISCUSSION THREAD

**Discussion by Andacanaver** [outline/outline#6628] [Ideas]
It would be nice to be able to have a slash command that would create a toggle, like in Notion, that could hide things. like the attached image, you could do a /toggle and it would create a toggle that would then be able to be opened/closed to hide or show the information. 
![2024-03-03 11_38_05-Window](https://github.com/outline/outline/assets/48130732/ecf815f5-0e20-4a99-8a9f-ad26fc94dbad)

**Comment by keithschacht** [comment]
I came here searching for the same feature. I thought maybe I was overlooking it within the product. I use this heavily in Notion and would love to have it in Outline.

The best implementation of expand/collapse is actually checklist. It was a little painful to move from Checkvist to Notion many years ago because Checkvist has implemented that feature so well, but that’s basically all it is. However, anyone thinking about implementing it for Outline should definitely try the Checkvist version. Their keyboard support is also really strong.

**Reply by tommoor** [reply]
I think this wouldn't be too complex to add these days, honestly easier than the existing collapsible headings. A collapsible list type might be a good start yea - perhaps the existing list could have a `collapsible: boolean` property to allow easily converting existing lists into collapsible ones

**Reply by keithschacht** [reply]
If you get serious about implementing it, I’d love to know. I have a bunch of very specific feedback I can share about how Notion’s implementing is lacking. I could write up some notes that may be helpful to you, and I’d be motivated by the chance it may make your implementation better (without being any more complex for users).

**Comment by thedigitaljournalist** [comment]
I want to give a serious +1 to this feature request - it’s one of the simplest and best features in Notion.

I’m loving Outline so far but definitely feel like this is the biggest feature gap.

**Reply by tommoor** [reply]
We will build it soon

**Reply by thedigitaljournalist** [reply]
Amazing!

**Reply by keithschacht** [reply]
@tommoor don't forget to try out checkvist before you do! It's an obscure product, but it is **the best** implementation of collapsible lists with amazing keyboard shortcut support. Notion's version is the first toggle lists most people have seen, but it falls short.

Actually, I recorded a quick demo video so it's easier for you to see: https://share.zight.com/d5uQ1Jng

**Reply by tommoor** [reply]
Appreciate the video – I think they are trying to accomplish different things though. One is a collapsible block of content (Notion) and the other is a nested task list

**Reply by keithschacht** [reply]
Oh, sorry, yes in that video I was just creating collapsible lists. That's how I most often use it. But Checkvist does equally well with collapsible blocks of content. After you've created a collapsible there is a different keyboard shortcut to give it contents. My keystrokes would be: "Type the title of my collapsible", press return, it defaults to assuming I'm going to create the next collapsible so I press Esc to cancel that, now the collapsible I just created is selected and I can add a note to it.

One core question is: which mode are you defaulting to?

When a user first invokes the command to create a collapsible (which is `>` in Notion's case) then is then should it (a) immediately default to letting the user type contents within that section, or (b) should it immediately default to letting the user create the **next** collapsible because they're later going to go back and fill them in.

For example, I most often use collapsible's when I'm writing an essay. I outline a list of points then I go back and fill in those points with paragraphs of text. But I do them as collapsible so I can very quickly see the outline of my argument and re-order key points. So my brain is: list first, fill out second. Whereas Notion defaults to assuming you want to fill out each one as soon as you create it. There is surely some way to have the best of both worlds.

**Reply by tommoor** [reply]
> One core question is: which mode are you defaulting to?

We'll see – but honestly Notion's implementation seems more fitting for a documentation product with the Checkvist version making sense for a list-based approach. The primary idea of these blocks is that you can collapse some content away by default when browsing docs

**Reply by keithschacht** [reply]
They can both live together simultaneously. But my suggestion is simply:

When you press return after entering your first toggle so that your state looks like this:

```
> First item
> 
```

It defaults for you to enter a second item. That’s how all other bullets work. But special to collapse, if you press backspace (to delete the carrot) then you’re entering contents for this first item.

Of course it’s up to you, but that’s my suggestion for how to improve over notion.
